src/lrs/io_schedulers/fifo.c

```
@@ -141,19 +142,116 @@ static int fifo_requeue(struct io_scheduler *io_sched,
     return 0;
 }
 
-static int fifo_peek_request(struct io_scheduler *io_sched,
-                             struct req_container **reqc)
+static int max_write_per_grouping(void)
 {
+    static int concurrent_write_per_grouping = -1;
+
+    if (concurrent_write_per_grouping == -1) {
+        concurrent_write_per_grouping =
+            PHO_CFG_GET_INT(cfg_lrs, PHO_CFG_LRS, fifo_max_write_per_grouping,
+                            0);
+        if (concurrent_write_per_grouping < 0) {
+            pho_error(-EINVAL,
```

major: if an error occurs here (which it can't with the previous comment), we should return an error instead, as is done in other parts of the code

---

This condition is not here to deal with a parsing error but only with a negative value written in the configuration file.

---

You're right on the parsing issue, but I still think we should return an error here instead

---

Done